app/infra/axios/axios-http-client.ts

```diff
@@ -1,13 +1,21 @@
-import { HttpPostClient } from "@ns/data/protocols/http-post-client";
-import AxiosHelper from "./axios-helper";
-import axios from "axios";
+import {
+  HttpClient,
+  HttpRequest,
+  HttpResponse,
+} from '@ns/data/protocols/http-client';
+import axios from 'axios';
 
-export class AxiosHttpClient implements HttpPostClient {
-  constructor (private url: string, private configurations?: any) { }
-
-  async post <P extends unknown, R extends unknown>(data: P): Promise<R> {
-    const formData = AxiosHelper.toFormData(data)
-    const response = await axios.post<P, R>(this.url, formData, this.configurations)
-    return response
+export class AxiosHttpClient implements HttpClient {
+  async request(data: HttpRequest): Promise<HttpResponse<any>> {
```

Comment:
Aqui não seria melhor tipar o retorno como HttpResponse<T> onde T é genérico? Assim possíveis erros de tipo poderiam ser evitados

Reply:
Sim, tem toda razão. Eu pensei em fazer um refactor depois para adicionar essa tipagem genérica, para não adicionar muita complexidade agora no início